Sources/CombineLongPolling/LongPollingPublisher.swift

```diff
@@ -104,10 +110,27 @@ extension LongPollingPublisher {
                                 self.semaphore.signal()
                             }
                         )
-                        .setFailureType(to: Never.self)
-                        .sink { result in
-                            _ = self.buffer?.buffer(value: result )
-                        }
+                        .sink(
+                            receiveCompletion: { [weak self] completion in
+                                guard let self = self else { return }
+
+                                guard case let .failure(error) = completion else {
+                                    // If this completes without error, we don't send completion to downstream, because the Long Polling will
+                                    // restart. Only error will stop the subscription and send a kill message to downstream.
+                                    return
+                                }
+
+                                self.started = false
```

Comment:
This access is not locked. Race condition with demand?

Reply:
This is locked by the semaphore. I assume that completion won't race against "start" and it's impossible to completion happens before the "start". So I'm locking start variable only before the beginning of the subscription, to avoid 2 demands racing against each other. I guess from that point on, we are pretty much safe and we only set start to false.

I'm a bit afraid of deadlock and avoiding locking too much unless I see a potential risk. But could be done.

Reply:
I totally agree on being cautions with locking. 

But I don't see any reason why completion and `request(demand:)` are safe from each other. Any subscriber can demand more, which will lock the lock and access `started`. We could totally at the same time finish our publisher and write `started` to false, resulting in corrupted access here. 

But I assume I am also currently writing the same bug..

Reply:
I'll review that. I think there are other potential problems in this "finishing" part. Currently the lock is only for protecting against double-start, not exactly the finishing part. But finishing indeed can collide with demand, so maybe we need a variable for "cancelled" as well.

Reply:
Ok, I think now it's more protected, please double-check if I'm not missing anything.